### PR TITLE
Fixed number of random reads by bcrypt cost 15

### PR DESCRIPTION
**Pull request by Sc00bz**
TL;DR
`randomReads = ((numberOfPboxInts * sizeOfInt + numberOfSboxInts * sizeOfInt) * (1 + 2 * 2 ** cost) + finishTimesEncrypted * finishMessageSize) / bytesPerBlockEncrypt * 4 * numberOfRounds`
`2,185,278,016 = ((18 * 4 + 256*4 * 4) * (1 + 2 * 2 ** 15) + 64 * 24) / 8 * 4 * 16`

Expand key calls encrypt `(numberOfPboxInts * sizeOfInt + numberOfSboxInts * sizeOfInt) / bytesPerBlockEncrypt` times. `521 = (18 * 4 + 256 * 4 * 4) / 8`
Encrypt does `4 * numberOfRounds` random reads. `64 = 4 * 16`
Expand key does `521 * 64` random reads. `33,344 = 521 * 64`
Expand key is called `1 + 2 * 2 ** cost` times. Cost=15: `65,537 = 1 + 2 * 2 ** 15`
Finishing calls encrypt `finishTimesEncrypted * finishMessageSize / bytesPerBlockEncrypt` times. `192 = 64 * 24 / 8`
All together that's `2,185,278,016 = 65,537 * 33,344 + 192 * 64` random reads.

I also modified an implementation of bcrypt to output the number of random reads and got the same answer. Also 1/4 is what I was expecting for Pufferfish when you count the random look ups to L2 cache (takes 2x as L1 cache) and count SHA512 operations.

### CODE REVIEW

**Comment by Sc00bz** [comment]
I didn't fix all the places this was referenced. The statement ", but performs many more reads than bcrypt for the same target runtimes" in the third feature should be removed. The harder to fix one is "From this example, it is overwhelmingly obvious to see that Pufferfish2 is much stronger than bcrypt, while providing much better GPU resistance than bcrypt." This needs rewording because it is overwhelmingly obvious that Pufferfish2 is much stronger than bcrypt, but it's not exactly obvious from the example.

Oh I should mention that I don't think I checked the Pufferfish2 numbers. I feel like I did awhile ago but not sure.
